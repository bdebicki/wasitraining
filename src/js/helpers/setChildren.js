export default function(element, elementChildren) {
	const el = element;
	const children = elementChildren;

	if (typeof (children) === 'string') {
		el.textContent = children;
	} else if (typeof (children) === 'object') {
		if (Array.isArray(children)) {
			children.forEach((childrenEl) => {
				if (typeof (childrenEl) === 'string') {
					el.append(childrenEl);
				} else if (childrenEl && typeof (childrenEl) !== 'string') {
					el.appendChild(childrenEl);
				}
			});
		} else {
			el.appendChild(children);
		}
	}
}
