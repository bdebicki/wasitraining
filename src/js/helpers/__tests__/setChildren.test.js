import using from '../../../../tests/utils/using';
import setChildren from '../setChildren';

const childrenCases = [
	{
		description: 'element with text',
		children: 'span element',
	},
	{
		description: 'element wit another element',
		children: document.createElement('span'),
	},
	{
		description: 'element with text and html element',
		children: [
			'span element',
			document.createElement('span'),
		],
	},
	{
		description: 'element with multiple html element',
		children: [
			document.createElement('span'),
			document.createElement('em'),
		],
	},
];

describe('tests for set children helper function', () => {
	using(childrenCases).describe('render <span> element with node', ({ description, children }) => {
		it(description, () => {
			// having
			const el = document.createElement('span');

			// when
			setChildren(el, children);

			// then
			expect(el).toMatchSnapshot();
		});
	});
});
