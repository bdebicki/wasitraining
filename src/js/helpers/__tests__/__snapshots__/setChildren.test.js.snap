// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`tests for set children helper function render <span> element with node element wit another element 1`] = `
<span>
  <span />
</span>
`;

exports[`tests for set children helper function render <span> element with node element with multiple html element 1`] = `
<span>
  <span />
  <em />
</span>
`;

exports[`tests for set children helper function render <span> element with node element with text 1`] = `
<span>
  span element
</span>
`;

exports[`tests for set children helper function render <span> element with node element with text and html element 1`] = `
<span>
  span element
  <span />
</span>
`;
